internal/core/src/exec/operator/groupby/SearchGroupByOperator.cpp

```diff
@@ -112,20 +115,150 @@ SearchGroupBy(const std::vector<std::shared_ptr<VectorIterator>>& iterators,
             break;
         }
         case DataType::VARCHAR: {
-            auto dataGetter =
-                GetDataGetter<std::string>(segment, group_by_field_id);
+            auto dataGetter = GetDataGetter<std::string, std::string>(
+                segment, group_by_field_id);
             GroupIteratorsByType<std::string>(iterators,
                                               search_info.topk_,
                                               search_info.group_size_,
                                               search_info.strict_group_size_,
-                                              *dataGetter,
+                                              dataGetter,
                                               group_by_values,
                                               seg_offsets,
                                               distances,
                                               search_info.metric_type_,
                                               topk_per_nq_prefix_sum);
             break;
         }
+        case DataType::JSON: {
+            AssertInfo(search_info.json_path_.has_value(),
+                       "json_path is required for json field when doing "
+                       "search_group_by");
+            if (search_info.json_cast_type_.has_value()) {
```

Comment:
is json_cast_type is set, then use json_cast_type to cast value to specific types
if not set, just stringnify any input